packages/amplify-gen2-codegen/src/index.ts

```diff
@@ -100,6 +102,12 @@ export const createGen2Renderer = ({
 
   const renderers: Renderer[] = [ensureOutputDir, ensureAmplifyDirectory, amplifyPackageJson, amplifyTsConfigJson, jsonRenderer];
 
+  if (unsupportedCategories && unsupportedCategories?.size >= 1) {
```

Comment:
nit: we can remove the `?` since we are checking for truthiness in the first part of `if`.